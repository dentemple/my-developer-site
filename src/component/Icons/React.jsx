// @flow

import React from 'react'
import styled, { keyframes } from 'styled-components'
import logo from './react-logo.svg'

const spin = keyframes`
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
`

const StyledImage = styled.img`
  animation: ${spin} infinite 20s linear;
  height: 80px;
`
const ReactLogo = () => <StyledImage src={logo} />

export default ReactLogo
