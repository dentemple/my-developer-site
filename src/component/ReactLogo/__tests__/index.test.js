import React from 'react'
import { shallow, render } from 'enzyme'
import MyComponent from '../index'

it('renders without crashing', () => {
  shallow(<MyComponent />)
})

it('matches the snapshot', () => {
  const component = render(<MyComponent />)
  expect(component).toMatchSnapshot()
})
