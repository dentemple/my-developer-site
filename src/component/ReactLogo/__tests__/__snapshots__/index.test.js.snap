// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
initialize {
  "0": Object {
    "attribs": Object {},
    "children": Array [
      Object {
        "attribs": Object {
          "class": "sc-bdVaJa cruMCr",
          "src": "logo.svg",
        },
        "children": Array [],
        "name": "img",
        "next": null,
        "parent": null,
        "prev": null,
        "root": [Circular],
        "type": "tag",
      },
    ],
    "name": "root",
    "next": null,
    "parent": null,
    "prev": null,
    "type": "root",
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xmlMode": false,
  },
}
`;
